src/test/java/io/github/enkarin/bookcrossing/books/service/BookServiceTest.java

```diff
@@ -0,0 +1,202 @@
+package io.github.enkarin.bookcrossing.books.service;
+
+import io.github.enkarin.bookcrossing.BookCrossingBaseTests;
+import io.github.enkarin.bookcrossing.books.dto.BookDto;
+import io.github.enkarin.bookcrossing.books.dto.BookFiltersRequest;
+import io.github.enkarin.bookcrossing.books.dto.BookModelDto;
+import io.github.enkarin.bookcrossing.exception.BookNotFoundException;
+import io.github.enkarin.bookcrossing.exception.UserNotFoundException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlGroup;
+
+import java.util.List;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static org.assertj.core.api.Java6Assertions.assertThatThrownBy;
+
+class BookServiceTest extends BookCrossingBaseTests {
+
+    @Autowired
+    private BookService bookService;
+
+    @AfterEach
+    void delete() {
+        jdbcTemplate.update("delete from t_book where book_id in (1, 2, 3, 4)");
```

Comment:
It won't work. You cannot bind to specific identifiers generated by the database in tests.

Reply:
2, 3 and 4 id set manually